webapp-client/src/component/Navbar.js

```diff
@@ -0,0 +1,93 @@
+import React, { Fragment } from 'react';
+// installation: npm i react-shadow-scroll or yarn add react-shadow-scroll
+import ReactShadowScroll from 'react-shadow-scroll';
+import '../stylesheets/Navbar.css';
+import Category from './Category';
+
+export default class Navbar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.default = <Category categoryName={'All'} />;
+    this.state = {
+      categories: {
+        All: this.default,
+      },
+      display: this.default,
+    };
+    this.addCategory = this.addCategory.bind(this);
+    this.setContent = this.setContent.bind(this);
+  }
+
+  addCategory(category) {
+    this.setState({
+      categories: {
+        ...this.state.categories,
+        [category]: <Category categoryName={category} />,
+      },
+    });
+  }
+
+  setContent(category) {
+    this.setState({
+      display: category,
+    });
+  }
+
+  render() {
+    return (
+      <Fragment>
+        <div className="nav-bar">
+          <CategoryList
+            categories={this.state.categories}
+            setContentPane={this.setContent}
+          />
+          <AddCategoryForm addCategory={this.addCategory} />
```

Comment:
Forgive me because I'm still learning react, but why is this a part of the navbar component?

Reply:
I feel like navbar is not the best name at this point, but the AddCategoryForm is at the bottom of navbar and changes the content of navbar, so I included it here.